critical_section.c

```diff
@@ -0,0 +1,24 @@
+#include <critical_section.h>
+#include <interrupts.h>
+#include <common.h>
+#include <libkern.h>
+
+volatile int cs_level = 0;
```

Comment:
Take a look at the following situation:
`cs_enter(); cs_leave(); cs_enter(); cs_leave();`.
What if the compiler decides to inline all of these calls and read cs_level only once? Between the two middle function calls, an interrupt might happen that changes the context. That new context might modify the cs_level value. I suppose that the compiler might do such an optimization and I didn't want to take chances.

BTW my initial idea was to use that cs_level variable every time interrupts are enabled/disabled, including the interrupt-handling routines.  Then it would be a debugging tool too.